docs/reference/cluster/get-desired-balance.asciidoc

```diff
@@ -0,0 +1,77 @@
+[[get-desired-balance]]
+=== Get desired balance API
+++++
+<titleabbrev>Get desired balance</titleabbrev>
+++++
+
+NOTE: {cloud-only}
+
+Exposes the desired balance and basic metrics.
+
+[[get-desired-balance-request]]
+==== {api-request-title}
+
+[source,console]
+--------------------------------------------------
+GET /_internal/desired_balance
+--------------------------------------------------
+// TEST[skip:Can't reliably test desired balance]
+
+The API returns the following result:
+
+[source,console-result]
+--------------------------------------------------
+{
+	"stats": {
+		"computation_active": false,
+		"computation_submitted": 5,
+		"computation_executed": 5,
+		"computation_converged": 5,
+		"computation_iterations": 4,
+		"computation_converged_index": 4,
+		"computation_time_in_millis": 0,
+		"reconciliation_time_in_millis": 0
+	},
+	"routing_table": {
+		"test": {
+			"0": {
+				"current": {
+					"state": "STARTED",
+					"primary": true,
+					"node": "UPYt8VwWTt-IADAEbqpLxA",
+					"node_is_desired": false,
+					"relocating_node": null,
+					"relocating_node_is_desired": false,
```

Comment:
I wonder if we want to show above 2 fields in case we do not reallocate? Both true/false in `relocating_node_is_desired`  are some what confusing if we are not relocating. If we always want to show all fields then may be we should consider null value if we are not relocating?

Reply:
I think it's generally good idea to keep all the fields in the JSON representations of objects which makes it easier for clients to consume them since they don't need to do additional check on the existence of the fields. 

To your second point, I kind of prefer to avoid three-valued logic for boolean fields, unless there's substantial semantic difference between `null` and `false`. I feel in this case if the relocating node is null, so it can't be desired, thus `relocating_node_is_desired` being `false` makes sense.